Source/WebKit/WebProcess/WebPage/CoordinatedGraphics/ThreadedCompositor.cpp

```diff
@@ -383,14 +380,24 @@ uint32_t ThreadedCompositor::requestComposition()
         m_attributes.viewportSize = webPage.size();
         m_attributes.deviceScaleFactor = webPage.deviceScaleFactor();
         m_attributes.viewportSize.scale(m_attributes.deviceScaleFactor);
+        ASSERT(!m_attributes.viewportSize.isEmpty());
         compositionRequestID = ++m_attributes.compositionRequestID;
     }
-    scheduleUpdate();
+    m_compositingRunLoop->scheduleUpdate();
     return compositionRequestID;
 }
 
 void ThreadedCompositor::scheduleUpdate()
 {
+    {
+        Locker locker { m_attributes.lock };
+        auto& webPage = m_layerTreeHost->webPage();
```

Comment:
```suggestion
        const auto& webPage = m_layerTreeHost->webPage();
```

`size()` and `deviceScaleFactor()` are both const, so we're OK to take a const reference to the web page